src/mpid/ch4/netmod/ofi/ofi_am.h

```diff
@@ -201,7 +201,10 @@ MPL_STATIC_INLINE_PREFIX bool MPIDI_NM_am_check_eager(MPI_Aint am_hdr_sz, MPI_Ai
         MPIDI_OFI_AMREQUEST(sreq, am_type_choice) = MPIDI_AMTYPE_SHORT;
         return true;
     } else {
-        if (MPIDI_OFI_ENABLE_RMA && !MPIR_CVAR_CH4_OFI_AM_LONG_FORCE_PIPELINE) {
+        if (MPIDI_OFI_ENABLE_TAGGED) {
+            /* We can support RNDV data using MPIDI_NM_am_tag_{send,recv} */
+            return false;
+        } else if (MPIDI_OFI_ENABLE_RMA && !MPIR_CVAR_CH4_OFI_AM_LONG_FORCE_PIPELINE) {
```

Comment:
Do we need some way to control the protocol here? Does this make it so RDMA read is never used if the provider supports tagged messaging? Also just reading the commit message, why don't we send an ack for the synchronous send when we start the RDMA read operation? Is there some limitation?

Reply:
Since this check is only from MPIDIG send, so only pt2pt am will skip RDMA read. RMA paths are not affected. The future plan is for `am_tag_send` to use `am_isend_long`, replacing the current "huge" and "striping" paths.

The plan is to leave it to the netmod to decide whether to use direct tag send or RDMA based on performance evaluations. And it will be orthogonal to the Eager/RNDV decisions.

The reason for omitting the sync ack for RDMA read (huge send) is because the huge send always do ack anyway, so the sync ack would be redundant.